### ISSUE TITLE
JVM Crash: Fullscreen doesnt work

### ISSUE THREAD

**Issue by JuKu** [libgdx/libgdx#5281]
#### Issue details

If i try to toggle fullscreen mode while playing, i get an JVM crash.

#### Reproduction steps/code

```java
Graphics.DisplayMode primaryMode = Gdx.graphics.getDisplayMode();
Gdx.graphics.setFullscreenMode(primaryMode);
```

#### Version of LibGDX and/or relevant dependencies
_Please provide the version(s) affected._

#### Stacktrace
```java
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ffcbadcaceb, pid=10456, tid=1724
#
# JRE version: Java(TM) SE Runtime Environment (8.0_91-b15) (build 1.8.0_91-b15)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.91-b15 mixed mode windows-amd64 compressed oops)
# Problematic frame:
# C  [glfw.dll+0xaceb]
#
# Failed to write core dump. Minidumps are not enabled by default on client versions of Windows
#
# An error report file with more information is saved as:
# D:\Users\Justin\IdeaProjects\island-exploration-rts\hs_err_pid10456.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
AL lib: (EE) alc_cleanup: 1 device not closed
```

[hs_err_pid10456.log](https://github.com/libgdx/libgdx/files/2134241/hs_err_pid10456.log)

#### Please select the affected platforms
- [ ] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [x] Windows
- [ ] Linux
- [ ] MacOS

**Comment by JuKu** [comment]
I have already found the solution:
`Gdx.input.isKeyJustPressed(Input.Keys.F)` fixed this issue.

**Comment by NathanSweet** [comment]
Maybe the render call is trying to avoid/reduce misshapen rendering when the window is resized?

**Comment by Darkyenus** [comment]
During normal window resizing, which happens through call to `GLFW.glfwPollEvents()`, normal rendering loop seems to stop, as it is blocked by `GLFW.glfwPollEvents()` for the duration of window resize dragging (at least on Mac). I don't know if this blocking is a bug or intended, but resizing the window triggers the resize callback, and the window never shows garbage (LWJGL2 did, IIRC), so it works well.

However, `setFullscreen/WindowedMode` triggers the resize callback immediately, which, as it is usually called inside `render()`, causes nested `render()` invocation. This is likely not desired behavior.

Two ways to fix this come to mind:
1. Disable the resize callback when `setFullscreen/WindowedMode` is called
2. Postpone the real window mode change so that it happens at the beginning of next frame.

Option 1. feels more intuitive, but could still lead to bugs, as application may not expect screen resolution to change mid-frame. The frame may then still show incorrectly resized frame, as it is rendering for the original screen size. 2. therefore is probably more correct. Without any modifications to resize callback, `render()` would still get called twice per frame. I don't know if GLFW expects `glfwSwapBuffers` to be called inside the callback, but if not, it may be worthwhile to disable the `render()` call from the resize callback when it happens during window mode change, so that it doesn't delay the whole operation.

**Comment by NathanSweet** [comment]
Noncontinuous rendering could be a problem if the frame is drawn with the wrong size. #2 sounds good to me.
